### ISSUE TITLE
[Feature request] context menu

### ISSUE THREAD

**Issue by MatiasMovie** [JasonStein/Notepads#287]
Add "Open with Notepads" in context menu.

**Comment by lindexi** [comment]
Do you means add the context menu in explorer? But I can not find how to do it

**Comment by JasonStein** [comment]
Basically you can write a simple registry batch file to associate few file extensions to hook up with an command. And for that command, it can be windows protocol like <notepads %filepath%>

**Comment by lindexi** [comment]
@JasonStein Do you means we should give a batch file to do it?

**Comment by JasonStein** [comment]
Yeah. UWP (without bridge) can not access registry. So batch file is the solution for average user.

**Comment by lindexi** [comment]
How about this code:

        <uap3:Extension Category="windows.fileTypeAssociation">
          <uap3:FileTypeAssociation Name="lindexi">
            <uap:SupportedFileTypes>
              <uap:FileType ContentType="foo/foo">.x</uap:FileType>
            </uap:SupportedFileTypes>
              <uap2:SupportedVerbs>
                  <uap3:Verb Id="doubi">逗比</uap3:Verb>
                  <uap3:Verb Id="Print">林德熙是逗比</uap3:Verb>
              </uap2:SupportedVerbs>
            <uap:DisplayName>林德熙逗比</uap:DisplayName>
            <uap:EditFlags OpenIsSafe="true"/>
          </uap3:FileTypeAssociation>
        </uap3:Extension>

![win10 uwp 关联文件1](https://user-images.githubusercontent.com/16054566/74619545-e6e25100-5170-11ea-8981-173f223f4c91.png)

**Comment by JasonStein** [comment]
@lindexi  This is for certain type/extension of files right? This is not a global command forwarder to trigger something like: Notepads %selectedFilePath%

**Comment by lindexi** [comment]
@JasonStein Yeah and it only support the extension that define in `<uap:FileType ContentType="foo/foo">.x</uap:FileType>`.

鉴于我的英文翻译太渣，还是说中文。 只有在 <uap:FileType ContentType="foo/foo">.x</uap:FileType> 里面定义的类型，并且点击打开方式选择了这个应用程序时，这个右键菜单才能生效。也许官方的期望是作为一个简便的入口，看了官方的例子是作为打开和打印等的菜单入口

换句话说，这个菜单不能用来所有的文件上，应该现在只有通过写注册表才能做到

**Comment by MarcAnt01** [comment]
Tbh, I don't understand how that would be useful, there's already "open in" menu from which you can select Notepads. If you want to avoid that just set it as default app. I like Desktop Bridge in general, but implementing for this purpose would make no sense

**Comment by revanmj** [comment]
> Tbh, I don't understand how that would be useful, there's already "open with" menu from which you can select Notepads. If you want to avoid that just set it as default app. I like Desktop Bridge in general, but implementing for this purpose would make no sense

This would be useful (if it worked for all file types) for extensions that Notepads do not advertise as supported - right now you can't use "Open with" menu for them (or at least I didn't find any way to make it work).

**Comment by JavierPAYTEF** [comment]
Not sure if I'm breaking a rule, but just want to chip in in favor of these feature since Notepad++ has it and I really miss it.

The feature is optional and can be enabled through the installer. It adds a global menu "Edit with Notepad++" as a Shell Extension of windows.
This is super useful, because I don't want to associate any file extensions to notepad++ (pretty much the same as regular Notepad), and in some cases it would be wrong to do that, for example if I want to open a file associated to another application, like html, css, js, etc.

Open With is a workaround, but it's really a hassle to go through the "Open With", looking for Notepads in the browser and finally opening the file.

There is also another useful thing to this feature, and it's to open a lot of files at once. You just select all the files, right click and select "Edit with Notepad++"

Here is a screenshot:
![image](https://user-images.githubusercontent.com/40425968/120884712-1c57bb80-c5bb-11eb-9ca5-a2b307d15fa1.png)
